### PR TITLE
Feat/attachments

### PR DESCRIPTION
**Pull request by georgmzimmer**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,6 +113,41 @@ Second, use the `HeraldPasswordResetForm` in place of django's built in `Passwor
...
+Sometimes you want to embed an image directly into the email content.  Do that by using a MIMEImage assigning a content id header to a MIMEImage, like this:
+
+    email = WelcomeEmail(user)
+    im = get_thumbnail(image_file.name, '600x600', quality=95)
+    my_image = MIMEImage(im.read()) # MIMEImage inherits from MIMEBase
+    my_image.add_header('Content-ID', '<{}>'.format(image_file.name))
+
+You can refer to these images in your html email templates using the Content ID (cid) like this:
+
+    <img src="cid:{{image_file.name}}" />
```
Comment (jproffitt) [inline README.md]: @georgmzimmer in order to have `image_file` in the template, you need to add that to the email context right? Should we mention that in the above example?

File: herald/base.py [hunk 2]
```diff
@@ -224,8 +237,17 @@ def get_extra_data(self):
...
+        Return a list of attachments or None.    
+
+        This only works with email.
+        """
+        return self.attachments
+
     @staticmethod
-    def _send(recipients, text_content=None, html_content=None, sent_from=None, subject=None, extra_data=None):
+    def _send(recipients, text_content=None, html_content=None, sent_from=None, subject=None, extra_data=None,
+              attachments=None):
```
Comment (jproffitt) [inline herald/base.py]: @georgmzimmer you added the attachments arg here, but you didn't add it to the BaseNotification class or the TwilioNotification class. They should all have the same method signature

File: herald/models.py [hunk 3]
```diff
@@ -35,6 +36,7 @@ class SentNotification(models.Model):
     status = models.PositiveSmallIntegerField(choices=STATUSES, default=STATUS_PENDING)
     notification_class = models.CharField(max_length=255)
     error_message = models.CharField(max_length=255, null=True, blank=True)
+    attachments = models.TextField(null=True)
```
Comment (jproffitt) [inline herald/models.py]: This should be `blank=True` as well I think. Or the user won't be able to save it in the admin without putting something in. I think...

File: requirements.txt [hunk 4]
```diff
@@ -1,4 +1,5 @@
 django>=1.8
 six
 mock
-twilio
\ No newline at end of file
+jsonpickle
```
Comment (jproffitt) [inline requirements.txt]: We need to add this requirements to the `install_requires` in setup.py too.

File: README.md (hunk 1 above)
Comment (georgmzimmer) [inline README.md]: Yes that would make sense.  It was just an example of how to get a thumbnailed image into an email.   You could just hardcode the filename too.  It just has to match what is in the Content-ID header.

File: herald/base.py (hunk 2 above)
Comment (georgmzimmer) [inline herald/base.py]: ok

File: herald/models.py (hunk 3 above)
Comment (georgmzimmer) [inline herald/models.py]: ok

File: requirements.txt (hunk 4 above)
Comment (georgmzimmer) [inline requirements.txt]: I didn't know about that. thanks

**Comment by jproffitt** [comment]
Thanks @georgmzimmer!

**Pull request merged**